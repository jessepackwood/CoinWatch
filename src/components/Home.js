import React, { Component } from 'react'
import { fetchCoinFront } from '../services/coinCapServices'
import Card from './Card'

class Home extends Component {
  constructor(){
    super()
  }

  componentDidMount() {
    fetchCoinFront()
  }

  render() {
    return (
      <div>
        <h1>Home</h1>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    coins: state.coins
  }
}

const mapDispatchProps = (dispatch) => {
  return {
    fetchCoinFront: () => {
      dispatch(fetchCoinFront)
    }
  }
}
export default Home;