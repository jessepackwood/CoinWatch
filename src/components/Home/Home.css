.home-page {
  padding: 30px; }

.home-title {
  text-align: center; }

.home-nav {
  display: flex;
  justify-content: center;
  font-size: 20pt; }
