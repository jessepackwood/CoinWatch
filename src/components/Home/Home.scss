
.home-page {
	padding: 30px;
}

.home-title {
  text-align: center;
}

.home-nav {
	.home-nav-link {
		// margin-right: 50px;
	}
	display: flex;
	justify-content: center;
	font-size: 20pt;
}