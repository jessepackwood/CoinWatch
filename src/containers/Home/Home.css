@import url("https://fonts.googleapis.com/css?family=Abel");
.home-page {
  padding: 30px; }

.home-page-subtitle {
  font-family: "Abel", sans-serif;
  display: inline-block;
  font-size: 18pt;
  margin-left: 10px;
  color: #fff; }

.home-title {
  text-align: center; }

.home-nav {
  display: flex;
  justify-content: center;
  font-size: 20pt; }

.filter-btn {
  height: 25px;
  width: 90px;
  margin: 0 5px;
  border-radius: 3px;
  outline: none;
  background: #262B2D;
  border: 2px solid orange;
  color: #fff; }
  .filter-btn:hover {
    cursor: pointer;
    border: 2px solid #9af765; }

.filter-btn-wrapper {
  margin-left: auto; }

.subtitle-wrapper {
  display: flex;
  justify-content: space-between; }

.spinner {
  height: calc(100vh - 110px);
  display: flex;
  justify-content: center;
  align-items: center;
  background: #262B2D; }
