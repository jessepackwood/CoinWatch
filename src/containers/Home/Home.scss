@import '../../variable.scss';

.home-page {
	padding: 30px;
}

.home-page-subtitle {
  display: inline-block;
  margin-left: 10px;
}

.home-title {
  text-align: center;
}

.home-nav {
	display: flex;
	justify-content: center;
	font-size: 20pt;
}

.filter-btn {
  height: 25px;
  width: 90px;
  margin: 0 5px;
  border-radius: 5px;
  border: 3px solid $blue;
}

.filter-btn-wrapper {
  margin-left: auto;
}

.subtitle-wrapper {
  display: flex;
  justify-content: space-between;
}

.spinner {
  margin-top: 200px;
  display: flex;
  justify-content: center;
  align-items: center;
}