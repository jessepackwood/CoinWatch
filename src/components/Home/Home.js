import React, { Component } from 'react'
import { connect } from 'react-redux'
import Header from '../Header/Header'
import Search from '../Search/Search'
import PropTypes from 'prop-types';
import './Home.css'

import Card from '../Card/Card'

class Home extends Component {
  constructor(props) {
    super()
    console.log(props)
    this.state = {
      coinsToDisplay: [],
      sortOrder: 'ascending',
      viewAll: false
    }
  }



  componentWillReceiveProps(nextProps) {
    if(nextProps.searchInput !== '') {
      return this.setState({coinsToDisplay: this.searchedCoinsToDisplay() })
    }
    this.setState({coinsToDisplay: nextProps.coins.slice(0,100)})

  }

  handleView = () => {
    if(!this.state.viewAll) {
      this.setState({viewAll: true, coinsToDisplay: this.props.coins})
    } else {
      this.setState({viewAll: false, coinsToDisplay: this.props.coins.slice(0,100)})
    }
  }

  handleSortClick = () => {
    if (this.state.sortOrder === 'ascending') {
      this.setState({sortOrder: 'descending', coinsToDisplay: this.props.coins.sort((a, b) => {
          return b.cap24hrChange - a.cap24hrChange
        }).slice(0, 100)
      })
    } else {
      this.setState({sortOrder: 'ascending', coinsToDisplay: this.props.coins.sort((a, b) => {
          return a.cap24hrChange - b.cap24hrChange
        }).slice(0, 100)
      })
    }
  }


  searchedCoinsToDisplay = () => {
    // debugger
      return this.props.coins.filter( coin => coin.long.toLowerCase().includes(this.props.searchInput.toLowerCase()))
  }
  // console.log(searchedCoinsToDisplay())
  render() {
    const mappedCoins = this.state.coinsToDisplay.map( (coin, index) => { 
      return <Card 
        coin={coin}
        key={`Card: ${index}`}
        number={`${index + 1}`}
      />
    })

    return (
      <div>
        <Header />
        <Search />
        <div className='home-page'>
          <h3>Top 100 Currencies by market cap</h3>
          <button onClick={this.handleView} > {this.state.viewAll ? 'Top 100' : 'View All'} </button>
          <button onClick={this.handleSortClick} >{`Sort: ${this.state.sortOrder}`}</button>
          {mappedCoins}
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  coins: state.coins,
  searchInput: state.searchInput.value
})

Home.propTypes = {
  coins: PropTypes.array
}

export default connect(mapStateToProps, null)(Home)