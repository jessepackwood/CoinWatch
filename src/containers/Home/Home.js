import React, { Component } from 'react'
import { connect } from 'react-redux'
import Header from '../Header/Header'
import Search from '../Search/Search'
import PropTypes from 'prop-types'
import './Home.css'
import MarketCapCard from '../MarketCapCard/MarketCapCard'
import { PulseLoader } from 'react-spinners'

export class Home extends Component {
  constructor() {
    super()
    this.state = {
      coinsToDisplay: [],
      sortOrder: 'Highest',
      viewAll: false,
      subtitle: 'Top 100 coins by market cap'
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.searchInput !== '') {
      return this.setState({coinsToDisplay: this.searchedCoinsToDisplay(nextProps.searchInput) })
    }
    this.setState({coinsToDisplay: nextProps.coins.slice(0, 100)})
  }

  handleView = () => {
    if (!this.state.viewAll) {
      this.setState({viewAll: true, subtitle: 'All coins listed by market cap', coinsToDisplay: this.props.coins})
    } else {
      this.setState({
        viewAll: false, subtitle: 'Top 100 coins by market cap', coinsToDisplay: this.props.coins.slice(0, 100)
      })
    }
  }

  handleSortClick = () => {
    if (this.state.sortOrder === 'Highest') {
      const highDayChangeCoins = this.props.coins
      this.setState({
        sortOrder: 'Lowest', 
        subtitle: 'Highest performing coins today',
        coinsToDisplay: highDayChangeCoins.sort((first, second) => {
          return second.cap24hrChange - first.cap24hrChange
        }).slice(0, 100)
      })
    } else {
      const lowDayChangeCoins = this.props.coins
      this.setState({
        sortOrder: 'Highest',
        subtitle: 'Lowest performing coins today',
        coinsToDisplay: lowDayChangeCoins.sort((first, second) => {
          return first.cap24hrChange - second.cap24hrChange
        }).slice(0, 100)
      })
    }
  }

  setToMarketCap = () => {
    const marketCapCoins = this.props.coins
    this.setState({
      subtitle: 'Top 100 coins by market cap',
      coinsToDisplay: marketCapCoins.sort((first, second) => {
      return second.mktcap - first.mktcap
      }).slice(0, 100)
    })
  }

  searchedCoinsToDisplay = (searchInput) => {
    return this.props.coins.filter( 
      coin => coin.long.toLowerCase()
        .includes(searchInput.toLowerCase())
    )
  }

  render() {
    const mappedCoins = this.state.coinsToDisplay.map( (coin, index) => { 
      return <MarketCapCard
        coin={coin}
        key={`Card: ${index}`}
        number={`${index + 1}`}
      />
    })

    return (
      <div>
        <Header />
        <Search />
        {this.props.coins.length === 0 && 
          <div className='spinner'>
            <PulseLoader 
              color={'orange'}
              size={25}
            />
          </div>
        }
        {this.props.coins.length > 0 &&
        <div className='home-page'>
          <div className='subtitle-wrapper'>
            <h3 className='home-page-subtitle'>
            {this.state.subtitle}
            </h3>
            <div className='filter-btn-wrapper'>
              <button 
                className='filter-btn view-button' 
                onClick={this.handleView} 
              > 
                {this.state.viewAll ? 'Top 100' : 'View All'} 
              </button>
              <button 
                className='filter-btn' 
                onClick={this.handleSortClick} 
              >
                {`Sort: ${this.state.sortOrder}`}
              </button>
              <button 
                className='filter-btn' 
                onClick={this.setToMarketCap} 
              >
              Market Cap
              </button>
            </div>
          </div>
          {mappedCoins}
        </div>
        } 
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  coins: state.coins,
  searchInput: state.searchInput.value
})

Home.propTypes = {
  coins: PropTypes.array,
  searchInput: PropTypes.string
}

export default connect(mapStateToProps, null)(Home)