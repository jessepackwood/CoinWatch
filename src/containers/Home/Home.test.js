import { Home, mapStateToProps } from './Home'
import React from 'react'
import { shallow } from 'enzyme'

describe('Home tests', () => {
  let home
  let mockStore
  let result
  const mockFunc = jest.fn()
  const mockHandleView = jest.fn()
  const mockHandleMarketCap = jest.fn()
  const mockCoins = [{long: 'bit', mktcap: 100, cap24hrChange: 4}, {long: 'iot', mktcap: 200, cap24hrChange: 1} ]

  beforeEach(()=> {
    mockStore = {
        coins: ['bit'],
        searchInput: ''
      };
    result = mapStateToProps(mockStore);

    home = shallow(<Home
                      coins={mockCoins}
                      searchInput={''}
                      store={mockStore}
                    />)
    });

  it('should be defined', () => {
    console.log(home.state())
    expect(home).toBeDefined()
  })
	
  it('should match the snapshot', () => {
    expect(home).toMatchSnapshot()
  })

  it('should have a default state', () => {
    const expectedDefault = {
      coinsToDisplay: [], 
      sortOrder: 'Highest', 
      viewAll: false
    };
    expect(home.state()).toEqual(expectedDefault);
  });

  it('should fire handleview', () => {
    home.instance().handleView()
    expect(home.state('viewAll')).toEqual(true)
  })

  it('should fire handleSortClick', () => {
    home.instance().handleSortClick()
    expect(home.state('sortOrder')).toEqual('Lowest')
  })

  it('should fire setToMarketCap', () => {
    home.instance().setToMarketCap()
    expect(home.state('coinsToDisplay').length).toEqual(2)
  })

})