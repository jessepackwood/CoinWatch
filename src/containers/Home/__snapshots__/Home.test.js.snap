// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    coins={
        Array [
            Object {
              "cap24hrChange": 4,
              "long": "bit",
              "mktcap": 100,
            },
            Object {
              "cap24hrChange": 1,
              "long": "iot",
              "mktcap": 200,
            },
          ]
    }
    searchInput=""
    store={
        Object {
            "coins": Array [
              "bit",
            ],
            "searchInput": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Connect(Component) />,
        false,
        <div
          className="home-page"
>
          <div
                    className="subtitle-wrapper"
          >
                    <h3
                              className="home-page-subtitle"
                    >
                              Top 100 Currencies by market cap
                    </h3>
                    <div
                              className="filter-btn-wrapper"
                    >
                              <button
                                        className="filter-btn view-button"
                                        onClick={[Function]}
                              >
                                        View All
                              </button>
                              <button
                                        className="filter-btn"
                                        onClick={[Function]}
                              >
                                        Sort: Highest
                              </button>
                              <button
                                        className="filter-btn"
                                        onClick={[Function]}
                              >
                                        Market Cap
                              </button>
                    </div>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="subtitle-wrapper"
>
              <h3
                            className="home-page-subtitle"
              >
                            Top 100 Currencies by market cap
              </h3>
              <div
                            className="filter-btn-wrapper"
              >
                            <button
                                          className="filter-btn view-button"
                                          onClick={[Function]}
                            >
                                          View All
                            </button>
                            <button
                                          className="filter-btn"
                                          onClick={[Function]}
                            >
                                          Sort: Highest
                            </button>
                            <button
                                          className="filter-btn"
                                          onClick={[Function]}
                            >
                                          Market Cap
                            </button>
              </div>
</div>,
            Array [],
          ],
          "className": "home-page",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="home-page-subtitle"
>
                  Top 100 Currencies by market cap
</h3>,
                <div
                  className="filter-btn-wrapper"
>
                  <button
                                    className="filter-btn view-button"
                                    onClick={[Function]}
                  >
                                    View All
                  </button>
                  <button
                                    className="filter-btn"
                                    onClick={[Function]}
                  >
                                    Sort: Highest
                  </button>
                  <button
                                    className="filter-btn"
                                    onClick={[Function]}
                  >
                                    Market Cap
                  </button>
</div>,
              ],
              "className": "subtitle-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Top 100 Currencies by market cap",
                  "className": "home-page-subtitle",
                },
                "ref": null,
                "rendered": "Top 100 Currencies by market cap",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="filter-btn view-button"
                      onClick={[Function]}
>
                      View All
</button>,
                    <button
                      className="filter-btn"
                      onClick={[Function]}
>
                      Sort: Highest
</button>,
                    <button
                      className="filter-btn"
                      onClick={[Function]}
>
                      Market Cap
</button>,
                  ],
                  "className": "filter-btn-wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "View All",
                      "className": "filter-btn view-button",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "View All",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sort: Highest",
                      "className": "filter-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Sort: Highest",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Market Cap",
                      "className": "filter-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Market Cap",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Connect(Component) />,
          false,
          <div
            className="home-page"
>
            <div
                        className="subtitle-wrapper"
            >
                        <h3
                                    className="home-page-subtitle"
                        >
                                    Top 100 Currencies by market cap
                        </h3>
                        <div
                                    className="filter-btn-wrapper"
                        >
                                    <button
                                                className="filter-btn view-button"
                                                onClick={[Function]}
                                    >
                                                View All
                                    </button>
                                    <button
                                                className="filter-btn"
                                                onClick={[Function]}
                                    >
                                                Sort: Highest
                                    </button>
                                    <button
                                                className="filter-btn"
                                                onClick={[Function]}
                                    >
                                                Market Cap
                                    </button>
                        </div>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="subtitle-wrapper"
>
                <h3
                                className="home-page-subtitle"
                >
                                Top 100 Currencies by market cap
                </h3>
                <div
                                className="filter-btn-wrapper"
                >
                                <button
                                                className="filter-btn view-button"
                                                onClick={[Function]}
                                >
                                                View All
                                </button>
                                <button
                                                className="filter-btn"
                                                onClick={[Function]}
                                >
                                                Sort: Highest
                                </button>
                                <button
                                                className="filter-btn"
                                                onClick={[Function]}
                                >
                                                Market Cap
                                </button>
                </div>
</div>,
              Array [],
            ],
            "className": "home-page",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="home-page-subtitle"
>
                    Top 100 Currencies by market cap
</h3>,
                  <div
                    className="filter-btn-wrapper"
>
                    <button
                                        className="filter-btn view-button"
                                        onClick={[Function]}
                    >
                                        View All
                    </button>
                    <button
                                        className="filter-btn"
                                        onClick={[Function]}
                    >
                                        Sort: Highest
                    </button>
                    <button
                                        className="filter-btn"
                                        onClick={[Function]}
                    >
                                        Market Cap
                    </button>
</div>,
                ],
                "className": "subtitle-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Top 100 Currencies by market cap",
                    "className": "home-page-subtitle",
                  },
                  "ref": null,
                  "rendered": "Top 100 Currencies by market cap",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <button
                        className="filter-btn view-button"
                        onClick={[Function]}
>
                        View All
</button>,
                      <button
                        className="filter-btn"
                        onClick={[Function]}
>
                        Sort: Highest
</button>,
                      <button
                        className="filter-btn"
                        onClick={[Function]}
>
                        Market Cap
</button>,
                    ],
                    "className": "filter-btn-wrapper",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "View All",
                        "className": "filter-btn view-button",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "View All",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sort: Highest",
                        "className": "filter-btn",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Sort: Highest",
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Market Cap",
                        "className": "filter-btn",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Market Cap",
                      "type": "button",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
